src/frontend/http.rs

```diff
@@ -25,18 +27,43 @@ use crate::{
 async fn write_file_handler(pk: &str, body: BodyStream, name: Option<String>) -> Result<String> {
     let pk = &Secp256k1PubKey::try_from(pk)?;
 
+    let is_mime_type = "default_mime_type".to_string();
+
     let file_stream: FileStream = stream::try_unfold(
-        (body, BytesMut::with_capacity(SEGMENT_SIZE * 2)),
-        |(mut body_stream, mut remainder)| async {
+        (
+            body,
+            BytesMut::with_capacity(SEGMENT_SIZE * 2),
+            is_mime_type.clone(),
+        ),
+        |(mut body_stream, mut remainder, mut is_mime_type)| async move {
             while remainder.len() < SEGMENT_SIZE {
                 if let Some(chunk) = body_stream.next().await {
                     let bytes = chunk?;
+
+                    let mut buffer = Vec::new();
+                    buffer.extend_from_slice(&bytes);
+
+                    // is_mime_type = "test_mime_type".to_string();
+
+                    let kind = infer::get(&buffer); // Use the extended buffer here
+                    match kind {
+                        Some(mime_type) => {
+                            is_mime_type = mime_type.mime_type().to_string();
+                        }
+                        None => debug!("no mime_type found"),
+                    };
+
+                    debug!(">>>>>>>>>>>> PASS TO TUPLE MIME_TYPE {}", &is_mime_type);
+
                     remainder.extend(bytes);
 
                     if remainder.len() >= SEGMENT_SIZE {
                         let segment = remainder.split_to(SEGMENT_SIZE);
                         trace!("Stream 1MB segment");
-                        return Ok(Some((segment.freeze(), (body_stream, remainder))));
+                        return Ok(Some((
+                            segment.freeze(),
```

Comment:
```suggestion
                            (segment.freeze(), mime_type)
```
This may be enough to pass the mime_type into fs using the FileStream